quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/FeaturesDistTest.java

```diff
@@ -3,22 +3,71 @@
 import io.quarkus.test.junit.main.Launch;
 import io.quarkus.test.junit.main.LaunchResult;
 import org.hamcrest.CoreMatchers;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
 import org.keycloak.it.junit5.extension.CLIResult;
 import org.keycloak.it.junit5.extension.DistributionTest;
+import org.keycloak.it.junit5.extension.RawDistOnly;
+import org.keycloak.quarkus.runtime.cli.command.Build;
+import org.keycloak.quarkus.runtime.cli.command.Start;
 import org.keycloak.quarkus.runtime.cli.command.StartDev;
 
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
 
 @DistributionTest
+@RawDistOnly(reason = "Containers are immutable")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 public class FeaturesDistTest {
 
+    @Test
+    @Launch({ Build.NAME, "--features=preview", "--cache=local"})
+    @Order(1)
+    public void testEnableOnBuild(LaunchResult result) {
+        CLIResult cliResult = (CLIResult) result;
+        cliResult.assertBuild();
+        assertPreviewFeaturesEnabled(cliResult);
+    }
+
+    @Test
+    @Launch({ Start.NAME, "--http-enabled=true", "--hostname-strict=false"})
+    @Order(2)
+    public void testFeatureEnabledOnStart(LaunchResult result) {
+        assertPreviewFeaturesEnabled((CLIResult) result);
+    }
+
     @Test
     @Launch({StartDev.NAME, "--features=preview"})
-    public void testPreviewFeaturesGetEnabledWhenCliArgIsSet(LaunchResult result) {
+    public void testEnablePreviewFeatures(LaunchResult result) {
         CLIResult cliResult = (CLIResult) result;
         cliResult.assertStartedDevMode();
+        assertPreviewFeaturesEnabled((CLIResult) result);
+    }
+
+    @Test
+    @Launch({StartDev.NAME, "--features=preview", "--features-disabled=token-exchange"})
+    public void testEnablePrecedenceOverDisable(LaunchResult result) {
+        CLIResult cliResult = (CLIResult) result;
+        cliResult.assertStartedDevMode();
+        assertPreviewFeaturesEnabled((CLIResult) result);
+    }
+
+    @Test
+    @Launch({StartDev.NAME, "--features=token-exchange,admin-fine-grained-authz"})
+    public void testEnableMultipleFeatures(LaunchResult result) {
+        CLIResult cliResult = (CLIResult) result;
+        cliResult.assertStartedDevMode();
+        assertThat(cliResult.getOutput(), CoreMatchers.allOf(
+                containsString("Preview feature enabled: admin_fine_grained_authz"),
+                containsString("Preview feature enabled: token_exchange")));
```

Comment:
kind of weird that here we use underscores again instead of the dash notation. Can you change this to also use dash notation instead pls?

Reply:
Agree is not aligned with Dist.X values. I don't think we should change this now. It is a message from the Profile class.